### PR TITLE
Replace `is_same<>::value` with `is_same_v<>`

### PR DESCRIPTION
**Pull request by codereport**
Simple PR that ran a regex to replace `is_same<...>::value` with `is_same_v<...>`

### CODE REVIEW

File: cpp/src/io/orc/orc.h [hunk 1]
```diff
@@ -197,23 +197,21 @@ class ProtobufReader {
...
+  template <typename base_t, typename std::enable_if_t<std::is_same_v<base_t, double>>* = nullptr>
   int static constexpr encode_field_number_base(int field_number) noexcept
   {
     return (field_number * 8) + PB_TYPE_FIXED64;
   }
 
   template <typename T,
-            typename std::enable_if_t<!std::is_class<T>::value or
-                                      std::is_same<T, std::string>::value>* = nullptr>
+            typename std::enable_if_t<!std::is_class<T>::value or std::is_same_v<T, std::string>>* =
```
Comment (mythrocks) [inline cpp/src/io/orc/orc.h:213]: There might be value in a follow-up PR to convert these SFINAE checks into `CUDF_ENABLE_IF()`.

**Review by mythrocks — APPROVED** [review]

File: cpp/src/io/csv/datetime.cuh [hunk 2]
```diff
@@ -419,7 +419,7 @@ __inline__ __device__ int64_t to_time_delta(char const* begin, char const* end)
   auto const second = parse_optional_integer<int8_t>(&cur, end, sep);
 
   int nanosecond = 0;
-  if (std::is_same<T, cudf::duration_s>::value) {
+  if (std::is_same_v<T, cudf::duration_s>) {
```
[the remark this answers is not in the document]
Comment (codereport) [reply cpp/src/io/csv/datetime.cuh:422]: Yea, there are probably a bunch of places that could benefit from that.

**Review by ttnghia — APPROVED** [review]

**Comment by codereport** [comment]
rerun tests

File: cpp/src/io/orc/orc.h (hunk 1 above)
Comment (codereport) [reply cpp/src/io/orc/orc.h:213]: Once we get clang-tidy set up we could definitely automate this.

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**